### ISSUE TITLE
Turnkey Linux cannot be installed via usb when Rufus used

### ISSUE THREAD

**Issue by Saentist** [pbatard/rufus#1197]
Checklist
---------
- [x] I looked at https://github.com/pbatard/rufus/wiki/FAQ to see if my question has already been answered.
- [x] I performed a search in the issue tracker for similar issues using keywords relevant to my problem, such as the error message I got from the log.
- [x] I clicked the "Log" button in Rufus (last small blue button before _START_) and copy/pasted the log into the line that says `<FULL LOG>` below. 
- [x] The log I am copying is the FULL log, starting with the line `Rufus version: x.y.z` - I have NOT removed any part of it.  

Additionally (if applicable):
- [x] I ran a bad blocks check, by clicking _Show advanced format options_ then _Check device for bad blocks_, and confirmed that my USB is not defective.
- [x] I also tried one or more of the following:
  - [x] Using a different USB drive.
  - [x] Plugging the USB into a different port.
  - [x] Running Rufus on a different computer.
- [x] If using an image, I clicked on the `(✓)` button to compute the MD5, SHA1 and SHA256 checksums, which are therefore present in the log I copied. I confirmed, by performing an internet search, that these values match the ones from the official image.

Issue description
-----------------

ISO to USB pass without any error's.
Boot on different PC also work to grub boot loader,
but live OS live starting or install search for cd/dvd.

![hzrsouu](https://user-images.githubusercontent.com/1206968/44308389-c8996a80-a3ac-11e8-9be9-ec4489bad6bd.jpg)

Used image:
http://mirror.turnkeylinux.org/turnkeylinux/images/iso/turnkey-core-15.0-stretch-amd64.iso
http://mirror.turnkeylinux.org/turnkeylinux/images/iso/turnkey-core-15.0-stretch-amd64.iso.hash
It is same situation with all images of TurnKey Linux
http://mirror.turnkeylinux.org/turnkeylinux/images/iso/
They are Debian based 

Log
---

```
Rufus version: 3.1.1320
Windows version: Windows 10 64-bit (Build 17134)
Syslinux versions: 4.07/2013-07-25, 6.03/2014-10-06
Grub versions: 0.4.6a, 2.02
System locale ID: 0x0402
Will use default UI locale 0x0402
SetLGP: Successfully set NoDriveTypeAutorun policy to 0x0000009E
Localization set to 'bg-BG'
NOTE: The Bulgarian (Български) translation requires an update, but the current translator hasn't submitted one. Because of this, some messages will only be displayed in English.
If you think you can help update this translation, please e-mail the author of this application
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x01595834
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: FAT32 (0x0b)
  Size: 3.7 GB (4009721856 bytes)
  Start Sector: 64, Boot: Yes
Scanning image...
ISO analysis:
  Image is an ISO9660 image
  Will use '/isolinux/isolinux.cfg' for Syslinux
  Detected Syslinux version: 6.03/20171018 (from '/isolinux/isolinux.bin')
Disk image analysis:
  Image has an unknown Master Boot Record
  Image is a bootable disk image
ISO label: 'core'
  Size: 251.4 MB (Projected)
  Note: File on disk is larger than reported ISO size by 228 KB...
  Uses: Syslinux/Isolinux v6.03
Using image: turnkey-core-15.0-stretch-amd64.iso (252 MB)

Computing checksum for 'D:\DROP\turnkey-core-15.0-stretch-amd64.iso'...
  MD5:	 3c8eede77e68367dd622895e4ed9f883
  SHA1:	 66207f2ecfc3cfeeeb6ff562d620be4330b7261c
  SHA256: 7c2d654313e934df8a03772680c34cbbc03d93ec6a208961e5f0b85b1c2f68f1

Downloading 'ldlinux.sys' from https://rufus.ie/files/syslinux-6.03/20171018/ldlinux.sys
Unable to access file: 404
Extended version was not found, trying main version...
Downloading 'ldlinux.sys' from https://rufus.ie/files/syslinux-6.03/ldlinux.sys
File length: 68599 bytes
Successfully downloaded 'ldlinux.sys'
Downloading 'ldlinux.bss' from https://rufus.ie/files/syslinux-6.03/20171018/ldlinux.bss
Unable to access file: 404
Extended version was not found, trying main version...
Downloading 'ldlinux.bss' from https://rufus.ie/files/syslinux-6.03/ldlinux.bss
File length: 512 bytes
Successfully downloaded 'ldlinux.bss'

Format operation started
Requesting disk access...
Opened \\.\PHYSICALDRIVE1 for exclusive write access
Requesting lock...
Will use 'G:' as volume mountpoint
I/O boundary checks disabled
Requesting lock...
Analyzing existing boot records...
Drive has a DOS/NT/95A (F2) Master Boot Record
Volume has an unknown Partition Boot Record
Clearing MBR/PBR/GPT structures...
Erasing 64 sectors
Deleting partitions...
Partitioning (MBR)...
Closing existing volume...
Waiting for logical drive to reappear...
Formatting (FAT32)...
Using cluster size: 4096 bytes
Quick format was selected
Creating file system...
Format completed.
Writing master boot record...
Drive has a Zeroed Master Boot Record
Partition is already FAT32 LBA...
Set bootable USB partition as 0x80
Using Syslinux MBR
Found volume GUID \\?\Volume{f87f68fd-a2dc-11e8-9c24-f46d0427b715}\
Installing Syslinux 6.03...
Opened \\?\Volume{f87f68fd-a2dc-11e8-9c24-f46d0427b715} for shared write access
Using existing './rufus_files/syslinux-6.03/20171018/ldlinux.sys' ✓
Using existing './rufus_files/syslinux-6.03/20171018/ldlinux.bss' ✓
Successfully wrote 'ldlinux.sys'
Successfully wrote Syslinux boot record
Successfully remounted \\?\Volume{f87f68fd-a2dc-11e8-9c24-f46d0427b715}\ on G:
Copying ISO files...
Extracting files...
Image is an ISO9660 image
This image will be extracted using Joliet extensions (if present)
Extracting: G:\casper\10root.squashfs (228.4 MB)
Extracting: G:\casper\initrd.gz (18.4 MB)
Extracting: G:\casper\vmlinuz (4 MB)
Extracting: G:\isolinux\boot.cat (2 KB)
Extracting: G:\isolinux\bootlogo (189.5 KB)
Extracting: G:\isolinux\chain.c32 (24.3 KB)
Extracting: G:\isolinux\gfxboot.c32 (10.3 KB)
Extracting: G:\isolinux\isolinux.bin (40 KB)
Extracting: G:\isolinux\isolinux.cfg (109 bytes)
Extracting: G:\isolinux\ldlinux.c32 (113.8 KB)
Extracting: G:\isolinux\libcom32.c32 (177.7 KB)
Extracting: G:\isolinux\menu.cfg (376 bytes)
Extracting: G:\isolinux\vesamenu.c32 (26.1 KB)
Created: G:\syslinux.cfg
Finalizing, please wait...
Created: G:autorun.inf
Created: G:autorun.ico

Found USB 2.0 device 'General USB Flash Disk USB Device' (090C:1000)
Using autorun.inf label for drive G: 'core'
1 device found
Disk type: Removable, Disk size: 4GB, Sector size: 512 bytes
Cylinders: 487, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x14DC14DA
Drive has a Syslinux Master Boot Record
Partition 1:
  Type: FAT32 LBA (0x0c)
  Size: 3.7 GB (4008706048 bytes)
  Start Sector: 2048, Boot: Yes
```

**Comment by pbatard** [comment]
Congratulations - By forcing me to edit your report because you still couldn't follow what the checklist asked you to do (is putting crosses into boxes or copy/pasting text, as you were explicitly asked to do, that hard?), you have have earned a __deferred__ tag, meaning that I am going to treat this issue as __super low__ priority and probably won't look into it for a few weeks.

**Comment by Saentist** [comment]
I report bug.
I not need to know BB code and tricks of github posting.

what is difference to write:
`I do all steps`
vs
`all this text and checks`

Even and in FAQ there is a мisspellings 

> mother tongue

From when muscles in the mouth = a vocal and written description of the words used by a nation?
Or in school kids learn French tongue not French language (except in cooking class)
 It is too easy to add criticism, but match harder to explain why.

**Comment by Saentist** [comment]
Creativity vs Bureaucracy = Evolution 
Or lets do right job not to chat unneeded.

p.s. 
it is not important how to say something, it is important other side to understand it.
I prefer short informative version
You prefer log Political version (99% air under pressure) nothing personal

**Comment by pbatard** [comment]
> You prefer log Political version

No. I prefer people who can follow the rules. It's really as simple as that and the root of all this "unneeded chat". The rules are precisely designed so that I don't have to waste more time than I need to fetching files off pastebin or wherever or not being able to see at one glance whether a step was completed or not.

You may think this is just wind, but maybe one day, once you have had to support __thousands__ of people through either an issue tracker or e-mail, you will understand why filtering issues according to who is willing to respect your requests, and who thinks they can just be ignored, is vital to being able to perform your job in the best conditions.

**Comment by pbatard** [comment]
After some testing, it appears that, like other maintainers before them, the Turnkey Linux maintainers seem to have chosen to deviate from what Debian and Ubuntu do, where they make sure they __don't restrict themselves to supporting the ISO-9660 file system only__, and dropped the ability for their __custom__ init script to look for data on FAT32 and other file systems, with the expected result that their script fail.

This is __NOT__ something Rufus can fix, as the issue is due to the distro maintainers being a bit too short-sighted with regards to the file system they support, and needing to modify their media detection script.

__HOWEVER__, once again I have to advise users to stop ignoring that big message that Rufus makes sure to display when it detects that an ISOHybrid is being used (which is the case here), and which is __REALLY THERE TO HELP YOU__ in case the image doesn't boot properly in ISO mode. This is the message that says:

![image1](https://user-images.githubusercontent.com/1206968/44926388-2765c900-ad49-11e8-88d7-33e905851830.png)

As the dialog __explicitly__ advises (in 38 different languages if it turns out that English is not your mother tongue):

> if you encounter issues during boot, you can try writing this image again in DD Image mode.

As it turns out, if you do just that, the image will boot without issues.

So, to summarize:

- The Turnkey people need to fix their media detection script to look for other file systems than ISO-9660. I'm not going to tell them how to do that since, if they follow what Debian and others do, they should be able to figure that out.
- Even then, this is not really an issue, as, if you do read and follow the advice that Rufus provides, you will create the USB in DD image mode, in which case the scripts will properly detect the media.

As such, since there is really nothing for me to address in Rufus, I will close this issue.
